### DISCUSSION TITLE
How to get new pid of sandbox?

### DISCUSSION THREAD

**Discussion by rieje** [netblue30/firejail#6271] [Q&A] [answered]
When I run weechat in a sandbox, its FIFO pipe is `/run/user/1000/weechat/weechat_fifo_9` where its FIFO pipe is in the format `/run/user/1000/weechat/weechat_fifo_<$PID>`. How can I get this PID of 9 for this process? `firejail --list` doesn't show this. I need it to get a predictable path for FIFO (weechat does not expose a way to configure this as a command line argument).

**Comment by ghost** [comment]
Have you tried `firejail --tree` yet to check if that shows what you need?

Alternatively, use a shell script:

```sh
#!/bin/sh
until _fifo="$(find "$XDG_RUNTIME_DIR" -maxdepth 1 -type p -name "weechat*")" > /dev/null; do
    sleep 0.5
done
# show me the FIFO
echo "$_fifo"
# run a weechat command
echo "foo" > "$_fifo" &
```

**Comment by rusty-snake** [comment, marked as answer]
`grep NStgid /proc/<PID>/status` should show `NStgid: <PID> 9`.

And see https://blogs.oracle.com/linux/post/translating-process-id-between-namespaces
